### PR TITLE
adds testhost package

### PR DESCRIPTION
**Pull request by rahku**

### CODE REVIEW

**Comment by rahku** [comment]
PTAL @gkhanna79 @weshaggard

File: src/.nuget/Microsoft.NETCore.TestHost/_.pdb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (weshaggard) [inline src/.nuget/Microsoft.NETCore.TestHost/_.pdb]: We should be sharing the same one for the entire repo.

File: src/.nuget/Microsoft.NETCore.TestHost/descriptions.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[
+    {
```
Comment (weshaggard) [inline src/.nuget/Microsoft.NETCore.TestHost/descriptions.json]: We should be using the shared version of this file for the entire repo.

File: src/.nuget/Microsoft.NETCore.TestHost/dir.props [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory).., dir.props))\dir.props" />
+
+  <!-- Packaging properties -->
+  <PropertyGroup>
+    <PackageDescriptionFile>$(MSBuildThisFileDirectory)descriptions.json</PackageDescriptionFile>
```
Comment (weshaggard) [inline src/.nuget/Microsoft.NETCore.TestHost/dir.props]: Shouldn't be needed once you share this file.

File: src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.builds [hunk 4]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (weshaggard) [inline src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.builds]: Can you please diff the nuspec's for this package with the ones we currently produce to make sure the match?

Comment (rahku) [inline src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.builds]: It looks similar to me. Will send you nupsec files

**Comment by rahku** [comment]
@dotnet-bot test OSX x64 Checked Build and Test
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by rahku** [comment]
ping!

File: build-packages.cmd [hunk 5]
```diff
@@ -60,6 +60,15 @@ if NOT [!ERRORLEVEL!]==[0] (
   exit /b 1
 )
 
+rem Build the TestHost package
+set __msbuildArgs="%__ProjectDir%\src\.nuget\Microsoft.NETCore.TestHost\Microsoft.NETCore.TestHost.builds" !allargs!
```
Comment (weshaggard) [inline build-packages.cmd]: Why does this need to be added directly to this cmd/sh script? I would expect them to just call build on packages.builds file.

**Comment by weshaggard** [comment]
LGTM other then the one comment about the scripts not calling build on packages.builds file which was not directly related to your change but is something we should change so we don't have to update these scripts everything we add a new package.

File: build-packages.cmd (hunk 5 above)
Comment (rahku) [inline build-packages.cmd]: I am not very sure why this script exists...but it had build commands for other package and I added testhost to it. @wtgodbe added this script.

Comment (wtgodbe) [inline build-packages.cmd]: I don't see any reason not to switch to calling build on packages.builds. @maririos and @Priya91 are making changes to these scripts right now, what do you guys think (build-packages is going to look very different as soon as Priya's change goes in anyways, so waiting might be the right move here)?

Comment (maririos) [inline build-packages.cmd]: Is the idea of this line to enforce building that package? packages.builds should be in charge of building all the packages.
As @wes points out, the script should only call one project and should avoid to have logic on it. With the changes Priya is doing we are going to be more close to that stage.
Independently of PR #6400, this change in the scripts should be avoided.

Comment (rahku) [inline build-packages.cmd]: this is already merged. Can you do the needful as part of PR #6400

Comment (maririos) [inline build-packages.cmd]: Oh you are right. @Priya91 could you please make sure we are building thios project the correct way? as we are building the other ones?

**Pull request merged**